src/pressure/VCycleMG.h

```diff
@@ -31,18 +29,35 @@ class VCycleMG: public IPressure{
     void Prolongate(Field &out, Field const &in, size_t level, bool sync = true);
     void Solve(Field &out, Field &tmp, Field const &b, size_t level, bool sync = true);
 
-    size_t m_levels;
-    size_t m_cycles;
-    size_t m_relaxs;
+    void call_smooth_colored_gauss_seidel(Field &out, Field &tmp, Field const &b, size_t level, bool sync);
+    void call_smooth_jacobi(Field &out, Field &tmp, Field const &b, size_t level, bool sync);
+    void call_solve_colored_gauss_seidel(Field &out, Field &tmp, Field const &b, size_t level, bool sync);
+    void call_solve_jacobi(Field &out, Field &tmp, Field const &b, size_t level, bool sync);
+
+    const size_t m_levels;
+    const int m_n_cycle;
+    const int m_n_relax;
+
+    void (VCycleMG::*m_smooth_diffusion_function)(Field &, Field &, Field const &, const size_t, bool);
+    void (VCycleMG::*m_solve_diffusion_function)(Field &, Field &, Field const &, const size_t, bool);
+    size_t m_diffusion_max_iter;
+    real m_diffusion_tol_res;
 
-    real m_dsign;
-    real m_w;
+    const real m_dt;
+    const real m_dsign = -1;
+    const real m_w;
 
-    std::vector<Field*> residuum0;
-    std::vector<Field*> residuum1;
-    std::vector<Field*> err0;
-    std::vector<Field*> error1;
-    std::vector<Field*> mg_temporal_solution;
+    Field **m_residuum0;
+    /**
+     * stores on level 0 rhs/b field
+     */
```

Comment:
why a multiline comment for a single line?

Reply:
30b2780